File: 프로그래머스/Lv.0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = 0
    
    while True:
        temp = arr[:] # 얇은 복사
        
        for i, num in enumerate(arr):
            if num >= 50 and not num % 2 :
                arr[i] = int(num / 2)
            elif num <= 50 and num % 2 :
                arr[i] = num * 2 + 1
        
        if temp == arr:
            break;
        else:
            temp = arr[:] # 얇은 복사
            answer += 1
    return answer

File: 프로그래머스/Lv.0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        int cnt = 0;
        
        while(answer == 0){
            
            int[] temp = Arrays.copyOf(arr, arr.length);
            for(int i = 0; i < arr.length; i++){
                arr[i] = arr[i] >= 50 ? (arr[i]%2==0 ? arr[i]/2 : arr[i]) : (arr[i]%2==0 ? arr[i] : arr[i]*2+1);
            }
            
            if(Arrays.equals(arr, temp)){
                break;
            }
            
            cnt++;
        }
        
        answer = cnt ;
        return answer;
    }
}